tests/topotests/isis_lfa_topo1/test_isis_lfa_topo1.py

```diff
@@ -979,14 +979,27 @@ def test_rib_ipv6_step24():
     tgen.net.cmd_raises("ip link set s1 down")
 
     rname = "rt1"
-    expect = '[{"multihop":true,"peer":"2001:db8:1000::2","local":"2001:db8:1000::1","status":"down"}]'
-    router_compare_json_output(
-        rname,
-        "show bfd peers json",
-        expect,
-        count=40,
-        wait=0.05,
-    )
+
```

Comment:
why are we writing a custom run and expect function?

Reply:
because when the bfd is down "show bfd peers json" returns [] and router_compare_json_output() with expected = [] never fails.

Reply:
you can pass a partial function to the run and expect to do the actual comparison you want.

Reply:
see _bgp_converge as an example in topotests, there are a bunch of them though